From fc5f57b2b00fff5a2f736c86fb3679492f14dfb8 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Mon, 21 Oct 2019 11:40:39 +0800
Subject: [PATCH] net: rockchip_wlan: rtl8723cs: work around clang bug

/tmp/rtl8703b_phycfg-53954c.s: Assembler messages:
/tmp/rtl8703b_phycfg-53954c.s:3071: Error: selected processor does not support `bfc w0,#4,#4'

Change-Id: I9a2cf7ea8b4da82ab6148043983ad57b68b93562
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 .../rockchip_wlan/rtl8723cs/hal/rtl8703b/rtl8703b_phycfg.c     | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rtl8723cs/hal/rtl8703b/rtl8703b_phycfg.c b/drivers/net/wireless/rockchip_wlan/rtl8723cs/hal/rtl8703b/rtl8703b_phycfg.c
index 6c1035f2bbb7..9a0799b29a2b 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8723cs/hal/rtl8703b/rtl8703b_phycfg.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8723cs/hal/rtl8703b/rtl8703b_phycfg.c
@@ -1044,6 +1044,7 @@ phy_GetSecondaryChnl_8703B(
 
 	}
 
+	barrier(); /* work around https://bugs.llvm.org/show_bug.cgi?id=42576 */
 	return (SCSettingOf40 << 4) | SCSettingOf20;
 }
 
@@ -1324,4 +1325,4 @@ PHY_SetSwChnlBWMode8703B(
 	PHY_HandleSwChnlAndSetBW8703B(Adapter, _TRUE, _TRUE, channel, Bandwidth, Offset40, Offset80, channel);
 
 	/* RTW_INFO("<==%s()\n",__FUNCTION__); */
-}
\ No newline at end of file
+}
-- 
2.35.3

